scipy/linalg/tests/test_lapack.py

```diff
@@ -1654,3 +1655,80 @@ def test_geqrfp_lwork(dtype, shape):
     m, n = shape
     lwork, info = geqrfp_lwork(m=m, n=n)
     assert_equal(info, 0)
+
+
+def generate_random_dtype_array(shape, dtype):
+    # generates a random matrix of desired data type of shape
+    if type(shape) != tuple:
+        # in case shape is not passed in as a tuple
+        shape = (shape,)
+    if dtype in COMPLEX_DTYPES:
+        return (np.random.rand(*shape)
+                + np.random.rand(*shape)*1.0j).astype(dtype)
+    return np.random.rand(*shape).astype(dtype)
+
+
+@pytest.mark.parametrize('dtype', DTYPES)
+@pytest.mark.parametrize('matrix_size', [(6, 7), (7, 6), (6, 6)])
+def test_geqrfp(dtype, matrix_size):
+    # set test tolerance appropriate for dtype
+
+    np.random.seed(42)
+    rtol = 250*np.finfo(dtype).eps
+    atol = np.finfo(dtype).eps
+    # get appropriate ?geqrfp for dtype
+    geqrfp = get_lapack_funcs(('geqrfp'), dtype=dtype)
+    gqr = get_lapack_funcs(("orgqr"), dtype=dtype)
+
+    np.random.seed(42)
+    m = matrix_size[0]
+    n = matrix_size[1]
+    # create random matrix of dimentions m x n
+    A = generate_random_dtype_array((m, n), dtype=dtype)
+    # create rq matrix using geqrfp
+    qr_A, tau, work, info = geqrfp(A)
+    # qr(A)
+    # obtain r from the upper triangular area
+    r = np.triu(qr_A)
+
+    # obtain q from the orgqr lapack routine:
+
+    if m > n:
+        # this adds an extra column to the end of qr_A
+        # let qqr be an empty m x m matrix
+        qqr = np.zeros((m, m), dtype=dtype)
+        # set first n columns of qqr to qr_A
+        qqr[:, :n] = qr_A
+        # determine q from this
+        q = gqr(qqr, tau=tau, lwork=m)[0]
+    else:
+        q = gqr(qr_A[:, :m], tau=tau, lwork=m)[0]
+
+    # test that q and r still make A
+    assert_allclose(q@r, A, rtol=rtol)
+    # enure that q is orthogonal (that q @ transposed q is the identity)
+    assert_allclose(np.eye(q.shape[0]), q@(q.conj().T), rtol=rtol,
+                    atol=atol*10)
```

Comment:
```suggestion
                    atol=atol)
```
I'd set `atol` to what you want it to be at the top.